### PR TITLE
Improve error reporting - Closes #3799

### LINKED ISSUE #3799: Improve transaction failed reporting
### Description

Improve transaction failed reporting for better error reporting which intern helps with addressing the issue

### Motivation

Whenever a user fail to create a transaction, the error report sent to `hubdev@lisk.io` does not contain useful information with regards to error. Having a useful information with regards to error is key to resolving the issue. 

### Additional information

**Current bug report**

Email bug

- Include error message along with error code

### PR DESCRIPTION
**Pull request by reyraa**
### What was the problem?

This PR resolves #3799

### How was it solved?

Added distinct error message for reclaim transaction errors.

### How was it tested?
By changing the transaction submission response and faking errors.

### CODE REVIEW

**Review by ManuGowda — COMMENTED** [review]
@reyraa I think the changes to the PR will not have an impact apart form reclaim transaction as its taking the static error from https://github.com/LiskHQ/lisk-desktop/blob/9cf9d3827c4ccb77119955af34acf177554ae417/src/components/shared/transactionResult/statusConfig.js#L32


And this is how the error report looks like 
```
Important metadata for the team, please do not edit:

   Lisk Core Version: 3.1, NetworkIdentifier: 4c09e6a781fc4c7bdb936ee815de8f94190f8a7519becd9de2081832be309a99, ServiceURL: https://service.lisk.com/

   Error Message: An error occurred while sending your transaction to the network. Please try again.

   Transaction: {"error":{"code":400},"transaction":{"moduleID":1000,"assetID":0,"senderPublicKey":"7310b042f97846b72ce55c6a6e40211470776b97a6c544b3f7aa6dac45d08f36","nonce":"0n","fee":"119000n","signatures":["e8797ca9be045970ebd2f7affc92670d529b74d84b40f1c236bf7d2950d674576bd5ad64b5c86d19927d553691457d806b31f520fdea8432ec7b147b19f07702"],"asset":{"amount":"4956714004n"},"id":"d9efeee307518cb08fe6506b0da0b8cf24629d7c1a14f5e32c9d64a9a97d5558"}}
```

Here, what we might need is actual error?

**Comment by ManuGowda** [comment]
@reyraa also is there nice way to show an info to users if they can not reclaim, we refer them to this blog post https://lisk.com/blog/announcement/lisks-holiday-security-reminder?

File: src/components/shared/transactionResult/regular.js [hunk 1]
```diff
@@ -38,6 +38,19 @@ const Regular = ({
     history.push(routes.wallet.path);
   };
 
+  const getReclaimError = (transaction) => {
+    if (transaction && transaction.assetID === 0 && transaction.moduleID === 2) {
```
Comment (ManuGowda) [inline src/components/shared/transactionResult/regular.js:42]: is the module id correct?

File: src/utils/helpers.js [hunk 2]
```diff
@@ -63,12 +63,13 @@ export const getErrorReportMailto = ({
...
   const recipient = 'desktopdev@lisk.com';
   const subject = `User Reported Error - Lisk - ${VERSION}`; // eslint-disable-line no-undef
+  const collisionAttackMessage = 'Failed to process reclaim your legacy balance. You might have been a victim of an address collision attack.';
   const body = encodeURIComponent(`
     \nImportant metadata for the team, please do not edit:
     \r
     Lisk Core Version: ${liskCoreVersion}, NetworkIdentifier: ${networkIdentifier}, ServiceURL: ${serviceUrl}
     \r
-    Error Message: ${errorMessage}
+    Error Message: ${error.indexOf('"moduleID":1000,"assetID":0') ? collisionAttackMessage : errorMessage}
```
Comment (ManuGowda) [inline src/utils/helpers.js:72]: use constant here too?

File: src/components/shared/transactionResult/regular.js [hunk 3]
```diff
@@ -38,6 +38,19 @@ const Regular = ({
     history.push(routes.wallet.path);
   };
 
+  const getReclaimError = () => {
+    if (typeof message === 'string' && message.indexOf('address collision attack') > -1) {
+      return (
+        <p>
+          <span>Read more about </span>
```
Comment (soroushm) [inline src/components/shared/transactionResult/regular.js:45]: could you please use the translation

**Review by soroushm — COMMENTED** [review]
init

**Review by ManuGowda — DISMISSED** [review]

**Review by soroushm — APPROVED** [review]

**Pull request merged**